### ISSUE TITLE
An "Action" to affect a Main ToolBar "Tool"when a message matches?

### ISSUE THREAD

**Issue by WoofGrrrr** [RealRaven2000/FiltaQuilla#297]
Would it be possible for the extension to provide some "Tool" that can be added to the Main Toolbar of the UI that can be affected by some new Filter Action?  Say, to display an indicator or something in this "Tool"?

I am trying to find a way to get something directly in the Thunderbird UI to indicate to me that I have NEW MESSAGES.  Someone in one of the forums I'm in suggested I try this FiltaQuilla extension, but I can't see how this can work for what I need without perhaps adding some new "Action".

### **SCENARIO:**

When Thunderbird is displaying some tab other than any Folder Tab, e.g. Calendar, Tasks, etc, the Folders List is not in view.  Thus I cannot see the total/unread/new message counts in the Folders List to know that I have new messages.

In this case, I don't see any visual cue directly in the Thunderbird UI that I have new messages.

I realize I can get alerts, system notifications, and/or a count on Thunderbird Taskbar Button or an indicator on the System Tray Icon, but I find alerts and notifications too distracting,  And amongst all the buttons on the Taskbar or icons in the System Tray, that one tiny count or indicator gets lost.  My eyes are not looking down there at the Taskbar.  My eyes are looking up.  I'm busy working with all my windows.

Just an Action to turn on some "Indicator" in this Toolbar "Tool" would suffice,  But it would be even better if this "Tool" could list the Total and/or Unread and/or New Message "Counts" for the Account.  Or maybe both.

### **WRONG EXTENSION?**

I understand that maybe this just doesn't fit the actual intent behind message "FILTERS".  For what I specifically need, the Match would probably always be "Match all messages" and "Apply filter when" has only "Getting New Mail" selected.

AND I don't have a clue how any New Message "Indicator" would get turned off or how the "Counts" would get updated as messages are Deleted, are no longer "Unread", or are just no longer "New".

Maybe this is extension - or Message Filters in general - is not the right solution for my problem.

But in the Forum I mentioned above, I was asking for an extension that would do what I need.  They suggested FiltaQuilla. That's why I'm here.  I thought I would give it a try.

I don't have the knowledge or skills to write an extension myself.  I know how to tweak the UI with userChrome.css and even unserChrome.js (using the userChromeJS extension,) but I have no idea how to have an extension get notified when message counts are updated and then to obtain those counts.

I have also submitted an Enhancement request into Bugzilla, but I'm sure it will get a very low priority if anyone ever gets a chance to even look at it.

**Comment by RealRaven2000** [comment]
Yeah I think this probably not something that a Filter should do - it should be in it's own dedicated extension. 

With an experiment it would be possible to write a toolbar  button that changes its color and includes and unread count when there a new email messages (biffstate changed) in all inboxes. For this one would need a folder listener that looks at the inboxes. So folder listeners just check changes in folders such as messages are added or removed. It is not quite the same as the message filters, which are supposed to process incoming messages.

Not sure if folder listeners are something that can also be implemented via APIs (to avoid experiments). With a standard toolbar button (for web extensions there is one button allowed per Add-on) we could probably change the label through the API, for instance adding the number of unread mails and maybe a colorful emoji to attract attention, e.g.:

☀️3 new, 15 unread

not sure if an API based (web extension) is also able to modify the icon itself. In experiment I simply add a class to the button (as I can interfere with UI directly and am not restricted to what the APIs offer) which can make the button look completely different. 

I use this technique in my Add-on QuickFolders for the quickMove function - it shows the number of mails I have dropped / queued for moving to another folder:

![image](https://github.com/user-attachments/assets/f204aa93-02d8-45ea-8d67-e07895f643bb)

here I just add a label (With the number of mails) to the icon and make it red to signify that these mails will be moved to the entered folder. With a plain web extension (standard mail extension) we are not allowed to modify the style of the label.

**Comment by WoofGrrrr** [comment]
Understood.

This started as a question in TB Add-on- Developers https://thunderbird.topicbox.com/groups/addons asking if there were a place where I could suggest an add-on that an extension developer might be interested in.

I described what I was looking for, and someone said I should check out FiltaQuilla.  After checking into it, I began to think it wasn't the solution.  But I thought I'd ask.

Thanks for responding!

**Comment by WoofGrrrr** [comment]
This will need an entirely different extension.

**Comment by RealRaven2000** [comment]
> Understood.
> 
> This started as a question in TB Add-on- Developers https://thunderbird.topicbox.com/groups/addons asking if there were a place where I could suggest an add-on that an extension developer might be interested in.

yeah I know, that was me actually. I thought your suggestion was closer to #240 which is something I want to eventually build...
